cmd/http/headers.go

```diff
@@ -126,6 +126,10 @@ const (
 
 	// Header indicates if the etag should be preserved by client
 	MinIOSourceETag = "x-minio-source-etag"
+	// Header indicates if the delete marker should be preserved by client
+	MinIOSourceDeleteMarker = "x-minio-source-deletemarker"
+	// Header indicates if the delete marker version needs to be purged.
+	MinIOSourceDeleteMarkerDelete = "x-minio-source-deletemarker-delete"
```

Comment:
how will this be used? Asking for my understanding.

Reply:
this is for deleting replicated delete-markers on the target cluster. If this flag is not passed, we would need additional overhead of a stat on each Delete call to determine if the object in question is a replica